app/src/main/java/net/squanchy/service/firebase/FirebaseAuthService.java

```diff
@@ -86,6 +86,7 @@ private Completable deleteUser(FirebaseUser user) {
 
     public Completable ifUserSignedInThenCompletableFrom(Func1<String, Completable> completableProvider) {
         return ifUserSignedIn()
+                .firstOrError()
```

Comment:
Makes sense to have it here, good catch! `ifUserSignedIn` is used also in the observable variant so it can't return an `Single`